wiki/Spotlight Guide.md

```diff
@@ -0,0 +1,142 @@
+      ) {
+        val viewState = when (feature) {
+          ONBOARDING_NEXT_BUTTON -> it.onboardingNextButton
+          TOPIC_LESSON_TAB -> it.topicLessonTab
+          TOPIC_REVISION_TAB -> it.topicRevisionTab
+          YOUR_FEATURE_NAME -> it.yourFeatureName
+```
+
+Similarly for the [recordSpotlightStateAsync](https://github.com/oppia/oppia-android/blob/d2c37dc547f3e5d12dfe62fa97b9b16fbf0fed6e/domain/src/main/java/org/oppia/android/domain/spotlight/SpotlightStateController.kt#L110) method.
+
+## Hooking up the spotlight
+In order to request a spotlight in the UI, use [SpotlightManager](https://github.com/oppia/oppia-android/blob/d2c37dc547f3e5d12dfe62fa97b9b16fbf0fed6e/app/src/main/java/org/oppia/android/app/spotlight/SpotlightManager.kt#L4):
+
+```kotlin
+checkNotNull(getSpotlightManager()).requestSpotlight(
+	SpotlightTarget(
+	binding.yourFeatureName,
+	R.string.your_feature_string_string_resource,
+	SpotlightShape.RoundedRectangle,
+	Spotlight.FeatureCase.YOUR_FEATURE_NAME
+)
+)
+
+  private fun getSpotlightManager(): SpotlightManager? {
+    return fragment.requireActivity().supportFragmentManager.findFragmentByTag(
+      SpotlightManager.SPOTLIGHT_FRAGMENT_TAG
+    ) as? SpotlightManager
+  }
+```
+
+For views that are laid out late in the lifecycle (such as recycler views loaded after a data provider call, view pagers, or elements that show up after an enter animation), use [requestSpotlightWithDelayedLayout](https://github.com/oppia/oppia-android/blob/d2c37dc547f3e5d12dfe62fa97b9b16fbf0fed6e/app/src/main/java/org/oppia/android/app/spotlight/SpotlightManager.kt#L13) instead.
+
+# Testing spotlights
+Spotlights are tested by checking if the expected hint is shown on the screen or not.
+
+```kotlin
+  @Test
+  fun testPromotedStorySpotlight_setToShowOnSecondLogin_notSeenBefore_checkSpotlightShown() {
+    logIntoUserTwice()
+    launch<HomeActivity>(createHomeActivityIntent(internalProfileId1)).use {
+      testCoroutineDispatchers.runCurrent()
+      onView(withText(R.string.promoted_story_spotlight_hint))
+        .check(matches(isDisplayed()))
+      // Or, use check(doesNotExist()) to verify it is not being shown.
+    }
+  }
+```
+
+In order to bypass a spotlight without any UI interaction, the ``SpotlightStateController`` should be used to mark the spotlights as seen. A helper function can be introduced and used to mark one or more spotlights as seen. 
+	
+```kotlin
+  private fun markAllSpotlightsSeen() {
+    val profileId = ProfileId.newBuilder().setInternalId(internalProfileId).build()
+    spotlightStateController.markSpotlightViewed(profileId, TOPIC_LESSON_TAB)  
+    spotlightStateController.markSpotlightViewed(profileId, TOPIC_REVISION_TAB) 
+    spotlightStateController.markSpotlightViewed(profileId, YOUR_FEATURE_NAME)
+    testCoroutineDispatchers.runCurrent()
+  }
+```
+
+## Disabling the spotlights
+If the spotlights need to be disabled, the platform parameter value should be set.
```

Comment:
This results in the reader asking the question: "Which platform parameter value?"

Either make the next couple of points bullet points, or make them separate paragraphs and use bold highlighting for the initial phrase (e.g. **In tests:** ..., **In production:** ...).

Reply:
done